### PR TITLE
Conversion event providers webpack partial

### PR DESCRIPTION
**Pull request by zutigrm**
## Summary

Addresses issue:

- #8531

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/event-providers/test-plugin-integration.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+addEventListener( 'DOMContentLoaded', () => {
```
Comment (tofumatt) [inline assets/js/event-providers/test-plugin-integration.js:1]: This looks like a test file that shouldn't be included in the plugin, right?

Let's move it out of this folder and put it in a test folder 😄 

Let's also add some comments explaining what it is/does/is for. Right now the purpose of this file is quite unclear! 😅

File: assets/js/event-providers/test-plugin-integration.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+	const array = [ 'test one', 'test two' ];
+
+	document.addEventListener( 'click', () => {
+		for ( const item in array ) {
+			// eslint-disable-next-line no-console
+			console.log( item );
```
Comment (tofumatt) [inline assets/js/event-providers/test-plugin-integration.js:7]:
```suggestion
			globa.console.log( item );
```

**Review by tofumatt — CHANGES_REQUESTED** [review]
Will this file be included in production builds? Let's not do that.

If you need to add steps to a QA brief to have the user test the webpack building of this it's fine, but I don't think we should be including test files like this in the repo 🤔

**Comment by zutigrm** [comment]
@tofumatt The purpose of the test file is just for QA:Eng, I described in the last step that this file should be deleted. I just scrambled some quick code, purely for testing the builds, and engineer testing it, will remove it

**Comment by tofumatt** [comment]
@zutigrm Yes, that makes sense to do. We shouldn't be checking in test code, especially code that will result in built files that would be shipped to production builds.

And removing it with a new PR would need to be tracked in a new issue for QA, etc. so I'll handle that part of the QA brief here and then remove the file 👍🏻

**Review by tofumatt — APPROVED** [review]

**Pull request merged**